p/scripts/main.js

```diff
@@ -256,45 +273,51 @@ function mark_favorite(div) {
 	}
 	pending_entries[div.id] = true;
 
-	$.ajax({
-		type: 'POST',
-		url: url,
-		data: {
-			ajax: true,
-			_csrf: context.csrf,
-		},
-	}).done(function (data) {
-		let inc = 0;
-		if (div.classList.contains('favorite')) {
-			div.classList.remove('favorite');
-			inc--;
-		} else {
-			div.classList.add('favorite');
-			inc++;
-		}
-		div.querySelectorAll('a.bookmark').forEach(function (a) { a.setAttribute('href', data.url); });
-		div.querySelectorAll('a.bookmark > .icon').forEach(function (img) { img.outerHTML = data.icon; });
+	const req = new XMLHttpRequest();
+	req.open('POST', url, true);
+	req.responseType = 'json';
+	req.onerror = function (e) {
+			openNotification(i18n.notif_request_failed, 'bad');
+			delete pending_entries[div.id];
+		};
+	req.onload = function (e) {
+			if (this.status != 200) {
+				return req.onerror(e);
+			}
+			const json = xmlHttpRequestJson(this);
+			let inc = 0;
+			if (div.classList.contains('favorite')) {
+				div.classList.remove('favorite');
+				inc--;
+			} else {
+				div.classList.add('favorite');
+				inc++;
+			}
+			div.querySelectorAll('a.bookmark').forEach(function (a) { a.setAttribute('href', json.url); });
```

Comment:
What's support for this like precisely?

Reply:
I do not understand the question 😕

Reply:
Sorry, it was a mobile post.

I thought not all modern(ish) browsers supported forEach yet but I guess I was thinking of IE8, so never mind. ;-)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

(I'd swear some older WebKit supported querySelector but not forEach, but since I'm clearly thinking of a decade ago figuring out if I recall correctly is not important.)

Reply:
Ah, see the Polyfills section at the top. 
I am trying to support IE11 (still easy enough so far) but not older.